File: Car Fleet.cpp
class Solution {
public:
    struct Car {
       int pos;
       double time;  
    };
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        vector<Car> cars(speed.size());
        int fleet = 0;
        for (int i = 0; i < position.size(); ++i)
              cars[i] = {position[i], (double)(target - position[i]) / speed[i]};
        sort(begin(cars), end(cars),
         [](const auto& a, const auto& b) { return a.pos > b.pos; });
        double max = 0;
        for (const auto& car : cars){
            if (car.time > max) {
                max = car.time;
                fleet++;
      }
        }
      return fleet;
    }
};

File: Car Fleet.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        cars = []
        answer = []

        for i in range(len(position)):
           cars.append((position[i], speed[i]))
        cars.sort()
        for c in cars:
            time = (target-c[0])/c[1]
            while answer and answer[-1] <= time:
                answer.pop()
            answer.append(time)
        return len(answer) 


        
        
        
